File: python/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        track = set(nums)
        longest = 0

        for i in track:
            if i - 1 not in track:
                length = 1
                while length + i in track:
                    length += 1
                longest = max(longest,length)
        
        return longest

File: javascript/easy/longest-consecutive-sequence.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let numMap = new Set(nums);
        let maxLen = 0;

        for(let num of nums){
            if(!numMap.has(num -1)){
                let len = 1;
                while(len < nums.length){
                    if(numMap.has(num + len)){
                        len++
                    }
                    else break;
                }
                maxLen = Math.max(maxLen,len);
            }
        }
        return maxLen;
    }
}
